internal/pkg/skuba/deployments/ssh/ssh.go

```diff
@@ -181,13 +192,19 @@ func (t *Target) internalSshWithStdin(silent bool, stdin string, command string,
 	return
 }
 
-func readerStreamer(reader io.Reader, outputChan chan<- string, description string, silent bool) {
+func readerStreamer(reader io.Reader, outputChan chan<- string, description string, silent bool, spin *spinner.Spinner) {
 	result := bytes.Buffer{}
 	scanner := bufio.NewScanner(reader)
 	for scanner.Scan() {
 		result.Write([]byte(scanner.Text()))
 		if !silent {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Stop()
+			}
 			klog.V(1).Infof("%s | %s", description, scanner.Text())
```

Comment:
Why not change to `klog.Infof` or `fmt.Printf`? Then we don't need spinner

Reply:
`klog.Infof` should not be used. `klog.V(x).Infof` should be used as this logs all `info` under a verbosity setting.

Reply:
Couldn't it be used for output where we want to respect the `klog` output settings, while printing with any verbosity level?

Reply:
This need to be verbosity level 1 because this includes details of ssh outputs. If we use Klog.Infof, we do not need verbosity level option in Skuba. We print all log output to customers. fmt.Print is not an option because Skuba output needs to be  consistent with timestamp. 
Spinner has simple task, giving an indicator to user while waiting. I saw, even with -v 1 that some of ssh cmds takes some times.

Reply:
I am not challenge spinner, but keep check the isTerminal in a for loop a kind of repeated job. It could be check outside for once.

Further, I am not sure stdout and stderr share the same spin, any corner case on this? Do we have some race condition unit tests on this?

Reply:
I am not challenge spinner, but keep check the isTerminal in a for loop a kind of repeated job. It could be check outside for once.

Further, I am not sure stdout and stderr share the same spin, any corner case on this? Do we have some race condition unit tests on this?

Reply:
Taking back my approval because it seems more concern on the thread safe even it seems the lib already taking care, but do agree to have adding unit tests to strengthen the implementation to avoid impact user experience. 

but thanks @suseclee keep improving this PR :)

Reply:
I do not think we need to worry about sharing because stdout and stderr also share same tty. I can store the value of isTerminal and use it instead of calling the function every time.

Reply:
I'm inclined to think that the spinner should only emit to stdout. Stderr should only be errors, and errors shouldn't need progress.

Reply:
The error is outcome of ssh cmd processes. We cannot expect which one will come out stderr or stdout.